internal/resources/aws/sns_topic.go

```diff
@@ -1,62 +1,230 @@
 package aws
 
 import (
+	"fmt"
 	"github.com/infracost/infracost/internal/resources"
 	"github.com/infracost/infracost/internal/schema"
+	"math"
 
 	"github.com/shopspring/decimal"
 )
 
 type SNSTopic struct {
-	Address         string
-	Region          string
-	RequestSizeKB   *float64 `infracost_usage:"request_size_kb"`
-	MonthlyRequests *int64   `infracost_usage:"monthly_requests"`
+	Address                 string
+	Region                  string
+	RequestSizeKB           *float64 `infracost_usage:"request_size_kb"`
+	MonthlyRequests         *int64   `infracost_usage:"monthly_requests"`
+	HTTPSubscriptions       *int64   `infracost_usage:"http_subscriptions"`
+	EmailSubscriptions      *int64   `infracost_usage:"email_subscriptions"`
+	KinesisSubscriptions    *int64   `infracost_usage:"kinesis_subscriptions"`
+	MobilePushSubscriptions *int64   `infracost_usage:"mobile_push_subscriptions"`
+	MacOSSubscriptions      *int64   `infracost_usage:"macos_subscriptions"`
+	SMSSubscriptions        *int64   `infracost_usage:"sms_subscriptions"`
+	SMSNotificationPrice    *float64 `infracost_usage:"sms_notification_price"`
 }
 
 var SNSTopicUsageSchema = []*schema.UsageItem{
 	{Key: "request_size_kb", ValueType: schema.Float64, DefaultValue: 0},
 	{Key: "monthly_requests", ValueType: schema.Int64, DefaultValue: 0},
+	{Key: "http_subscriptions", ValueType: schema.Int64, DefaultValue: 0},
+	{Key: "email_subscriptions", ValueType: schema.Int64, DefaultValue: 0},
+	{Key: "kinesis_subscriptions", ValueType: schema.Int64, DefaultValue: 0},
+	{Key: "mobile_push_subscriptions", ValueType: schema.Int64, DefaultValue: 0},
+	{Key: "macos_subscriptions", ValueType: schema.Int64, DefaultValue: 0},
+	{Key: "sms_subscriptions", ValueType: schema.Int64, DefaultValue: 0},
+	{Key: "sms_notification_price", ValueType: schema.Float64, DefaultValue: 0.0075},
+}
+
+// This is an experiment to see if using an explicit structure to define the cost components
+// can enable anything interesting (e.g. list what cost components could apply to a resource
+// without having any IaAC)
+func (r *SNSTopic) CostComponents() []*schema.CostComponent {
```

Comment:
I know this is marked as an experiment, but are we using this anywhere right now? It doesn't look like it, so I'd just remove it, otherwise it's just dead code.

Reply:
Fair, I'll comment it out so it's clear.